controllers/dscinitialization/servicemesh_test.go

```diff
@@ -0,0 +1,46 @@
+package dscinitialization_test
+
+import (
+	"github.com/go-logr/logr"
+
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/controllers/dscinitialization"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var defaultAudiences = []string{"https://kubernetes.default.svc"}
+
+var _ = Describe("Audiences", func() {
+	Describe("Determining if provided audiences are default", func() {
+		Context("When no specific audiences are provided", func() {
+			It("should consider the audiences to be default", func() {
+				var specAudiences *[]string
+				Expect(dsci.IsDefaultAudiences(specAudiences)).To(BeTrue())
+			})
+		})
+
+		Context("When the default audiences are provided", func() {
+			It("should consider these audiences to be default", func() {
+				specAudiences := &defaultAudiences
+				Expect(dsci.IsDefaultAudiences(specAudiences)).To(BeTrue())
+			})
+		})
+
+		Context("When non-default audiences are provided", func() {
+			It("should not consider these audiences to be default", func() {
+				specAudiences := &[]string{"https://example.com"}
+				Expect(dsci.IsDefaultAudiences(specAudiences)).To(BeFalse())
+			})
+		})
+	})
+
+	Describe("Determining the effective cluster audiences", func() {
+		Context("When non-default audiences are provided", func() {
+			It("should return the provided audiences", func() {
+				specAudiences := &[]string{"https://example.com"}
+				Expect(dsci.GetEffectiveClusterAudiences(specAudiences, nil, logr.Logger{})).To(Equal(*specAudiences))
+			})
+		})
+	})
+})
```

Comment:
If we move it to `cluster` pkg there's k8s envtest setup in place, so you will have access to the Kubernetes API server spawned by it.

This might give you different results, though I see you are passing `nil` to the client, which in fact means we only test faulty path here.

I think we should test `GetEffectiveClusterAudiences` only, and pass instance of fake client where we can easily define stubs - https://github.com/kubernetes-sigs/controller-runtime/tree/main/pkg/client/fake - specifically [sth like this](https://github.com/kubernetes-sigs/controller-runtime/blob/20f3f4bed925a9b7e41388c993fb217e7821910f/pkg/client/fake/client_test.go#L1373) can be helpful